app/src/sharedTest/java/org/oppia/app/testing/InputInteractionViewTestActivityTest.kt

```diff
@@ -288,9 +288,64 @@ class InputInteractionViewTestActivityTest {
   fun testFractionInputInteractionView_withInputtedLongNumber_clickSubmitButton_numberTooLongErrorIsDisplayed() {
     ActivityScenario.launch(InputInteractionViewTestActivity::class.java).use {
       onView(withId(R.id.test_fraction_input_interaction_view)).perform(typeText("12345678"))
-      closeSoftKeyboard ()
+      closeSoftKeyboard()
       onView(withId(R.id.submit_button)).check(matches(isDisplayed())).perform(click())
       onView(withId(R.id.fraction_input_error)).check(matches(withText(R.string.fraction_error_larger_than_seven_digits)))
     }
   }
+
+  @Test
+  fun testNumericInputInteractionView_withInputtedInvalidCharacter_invalidCharacterErrorIsDisplayed() {
+    ActivityScenario.launch(InputInteractionViewTestActivity::class.java).use {
+      onView(withId(R.id.test_number_input_interaction_view)).perform(typeText("/"))
+      onView(withId(R.id.number_input_error)).check(matches(withText(R.string.number_error_invalid_format)))
+    }
+  }
+
+  @Test
+  fun testNumericInputInteractionView_withInputtedLongNumber_clickSubmitButton_numberTooLongErrorIsDisplayed() {
+    ActivityScenario.launch(InputInteractionViewTestActivity::class.java).use {
+      onView(withId(R.id.test_number_input_interaction_view)).perform(typeText("12345678.6787687678"))
```

Comment:
Maybe also test the case where the number is too long without a decimal.

Reply:
Ok Done